pepper-apis/src/main/java/org/broadinstitute/ddp/event/pubsubtask/api/PubSubTaskMessageParser.java

```diff
@@ -0,0 +1,113 @@
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask.ATTR_TASK_TYPE;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask.ATTR_USER_ID;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskLogUtil.errorMsg;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskLogUtil.infoMsg;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.Map;
+
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.gson.Gson;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import org.broadinstitute.ddp.util.GsonUtil;
+import org.slf4j.Logger;
+
+/**
+ * Parses {@link PubSubTask} from subscription
+ * defined by config param "pubsub.pubSubTasksSubscription".
+ * The result {@link PubSubTaskMessageParseResult} contains
+ * created {@link PubSubTask} with parsed message.
+ * If error occured during parsing then {@link PubSubTaskMessageParseResult}
+ * contains non-null error message (to be sent to result topic).
+ */
+public class PubSubTaskMessageParser {
+
+    private static final Logger LOG = getLogger(PubSubTaskMessageParser.class);
+
+    private final ProjectSubscriptionName projectSubscriptionName;
+    private final PubSubTaskProcessorFactory pubSubTaskProcessorFactory;
+
+    private final Gson gson = GsonUtil.standardGson();
+
+    public PubSubTaskMessageParser(ProjectSubscriptionName projectSubscriptionName,
+                                   PubSubTaskProcessorFactory pubSubTaskProcessorFactory) {
+        this.projectSubscriptionName = projectSubscriptionName;
+        this.pubSubTaskProcessorFactory = pubSubTaskProcessorFactory;
+    }
+
+    public PubSubTaskMessageParseResult parseMessage(PubsubMessage message, AckReplyConsumer consumer) {
+        String messageId = message.getMessageId();
+        String taskType = message.getAttributesOrDefault(ATTR_TASK_TYPE, null);
+        String participantGuid = message.getAttributesOrDefault(ATTR_PARTICIPANT_GUID, null);
+        String userId = message.getAttributesOrDefault(ATTR_USER_ID, null);
+        String studyGuid = message.getAttributesOrDefault(ATTR_STUDY_GUID, null);
+        String payloadJson = message.getData() != null ? message.getData().toStringUtf8() : null;
+
+        LOG.info(infoMsg("Pubsub message received[subscription={}, id={}]: taskType={}, participantGuid={}, userId={}, "
+                        + "studyGuid={}, data={}"),
+                projectSubscriptionName, messageId, taskType, participantGuid, userId, studyGuid, payloadJson);
+
+        var pubSubTaskDescriptor = pubSubTaskProcessorFactory.getPubSubTaskDescriptors(taskType);
+
+        String errorMessage = null;
+        Object payloadObject = null;
+        PubSubTask.PayloadMap payloadMap = new PubSubTask.PayloadMap();
+        if (pubSubTaskDescriptor.isPayloadConvertibleToMap()) {
+            payloadMap.setMap(gson.fromJson(payloadJson, Map.class));
+        }
```

Comment:
What happens if this is not convertible to a map?  I think it's safe to assume the data is always going to be json, so let's just use json and dispense with the conversion to a map that requires fields to be strings.

Reply:
Agree. Fixed